pkg/tfbridge/provider_test.go

```diff
@@ -4535,3 +4535,339 @@ func TestPlanResourceChangeStateUpgrade(t *testing.T) {
+						Optional: true,
+						Elem:     &schemav2.Schema{Type: schemav2.TypeString},
+					},
+					"string_prop": {
+						Type:     schema.TypeString,
+						Optional: true,
+					},
+					"list_prop": {
+						Type:     schema.TypeList,
+						Optional: true,
+						Elem:     &schemav2.Schema{Type: schemav2.TypeString},
+					},
+					"nested_list_prop": {
+						Type:     schema.TypeList,
+						Optional: true,
+						Elem: &schemav2.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem:     &schemav2.Schema{Type: schemav2.TypeString},
+						},
+					},
+					"max_items_one_prop": {
+						Type:     schema.TypeList,
+						Optional: true,
+						MaxItems: 1,
+						Elem:     &schemav2.Schema{Type: schemav2.TypeString},
+					},
+				},
+			},
+		},
+	}
+	shimProv := shimv2.NewProvider(p, shimv2.WithPlanResourceChange(func(tfResourceType string) bool { return false }))
+	provider := &Provider{
+		tf:     shimProv,
+		config: shimv2.NewSchemaMap(p.Schema),
+		info: ProviderInfo{
+			P:              shimProv,
+			ResourcePrefix: "example",
+			Resources: map[string]*ResourceInfo{
+				"example_resource": {Tok: "ExampleResource"},
+			},
+		},
+	}
+	provider.initResourceMaps()
+
+	t.Run("unknown for string prop", func(t *testing.T) {
+		testutils.Replay(t, provider, `
+	{
+		"method": "/pulumirpc.ResourceProvider/Create",
+		"request": {
+			"urn": "urn:pulumi:dev::teststack::ExampleResource::exres",
+			"properties":{
+				"__defaults":[],
+				"string_prop":"04da6b54-80e4-46f7-96ec-b56ff0331ba9"
+			},
+			"preview":true
+		},
+		"response": {
+			"properties":{
+				"id":""
+			}
+		}
+	}`)
+	})
+
+	t.Run("unknown for set prop", func(t *testing.T) {
+		testutils.Replay(t, provider, `
+	{
+		"method": "/pulumirpc.ResourceProvider/Create",
+		"request": {
+			"urn": "urn:pulumi:dev::teststack::ExampleResource::exres",
+			"properties":{
+				"__defaults":[],
+				"setProps":["04da6b54-80e4-46f7-96ec-b56ff0331ba9"]
+			},
+			"preview":true
+		},
+		"response": {
+			"properties":{
+				"id":""
+			}
+		}
+	}`)
+	})
+
+	t.Run("unknown for list prop", func(t *testing.T) {
+		testutils.Replay(t, provider, `
+	{
+		"method": "/pulumirpc.ResourceProvider/Create",
+		"request": {
+			"urn": "urn:pulumi:dev::teststack::ExampleResource::exres",
+			"properties":{
+				"__defaults":[],
+				"listProps":["04da6b54-80e4-46f7-96ec-b56ff0331ba9"]
+			},
+			"preview":true
+		},
+		"response": {
+			"properties":{
+				"id": ""
+			}
+		}
+	}`)
+	})
+
+	t.Run("unknown for nested list prop", func(t *testing.T) {
+		testutils.Replay(t, provider, `
+	{
+		"method": "/pulumirpc.ResourceProvider/Create",
+		"request": {
+			"urn": "urn:pulumi:dev::teststack::ExampleResource::exres",
+			"properties":{
+				"__defaults":[],
+				"nestedListProps":[["04da6b54-80e4-46f7-96ec-b56ff0331ba9"]]
+			},
+			"preview":true
+		},
+		"response": {
+			"properties":{
+				"id":"",
+				"nestedListProps":["04da6b54-80e4-46f7-96ec-b56ff0331ba9"]
```

Comment:
What's going on here

Reply:
very curious but unrelated to the changes in this PR - I've verified the same behaviour on master